pkg/domain/user/repository/repository_usecase.go

```diff
@@ -11,9 +12,9 @@ type Repository struct {
 	DB *gorm.DB // this can be any gorm instance
 }
 
-func (r Repository) FindAll() (users *gorm.DB, err error) {
-	users = r.DB.Find(&users)
-	err = users.Error
+func (r Repository) FindAll() (users []user.User, err error) {
```

Comment:
```suggestion
// FindAll find the users in DB
func (r Repository) FindAll() (users []user.User, err error) {
```

Adicionar comentário para a função